packages/headless/src/utils/expression-builder-sapi-v4.ts

```diff
@@ -0,0 +1,187 @@
+type QueryExpresion =
+  | KeyWord
+  | ExactMatch
+  | FieldExpresion
+  | Near
+  | Not
+  | FieldExists
+  | ObjectAccess
+  | QueryExtensionInvocation
+  | QuerySyntax
+  | And
+  | Or;
+
+type FieldValue = DateRange | Date | KeyWord | ExactMatch;
+
+type TextValue = KeyWord | ExactMatch;
+
+interface DateRange {
+  type: 'dateRange';
+  dateRange: {
+    from: string;
+    to: string;
+  };
+}
+
+interface Date {
+  type: 'date';
+  dateValue: string;
+}
+
+interface KeyWord {
+  type: 'keyword';
+  value: string;
+}
+
+interface ExactMatch {
+  type: 'exactMatch';
+  value: string;
+}
+
+interface And {
+  type: 'and';
+  expressions: QueryExpresion[];
+}
+
+interface Or {
+  type: 'or';
+  expressions: QueryExpresion[];
+}
+
+interface FieldExpresion {
+  type: 'fieldExpression';
+  fieldName: string;
+  operator:
+    | 'contains'
+    | 'isExactly'
+    | 'lowerThan'
+    | 'lowerThanOrEqual'
+    | 'greaterThan'
+    | 'greaterThanOrEqual'
+    | 'phoneticMatch'
+    | 'fuzzyMatch'
+    | 'regexMatch'
+    | 'wildcardMatch'
+    | 'differentThan';
+  value: FieldValue; // shouldn't this be an array?
```

Comment:
@acote-coveo, it seems like `value` is not an array in swagger. Is this intended? or just a mistake in the doc?

Reply:
Ok, I was wondering if the intent was to specify multiple `fieldExpression` objects, one for each value. Supporting arrays would  help make the overall object less verbose 👍